include/revng/Pipeline/Location.h

```diff
@@ -0,0 +1,181 @@
+                                  RechabilityHelper<typename From::Parent, To>>;
+
+public:
+  static constexpr bool value = Found || Next::value;
+};
+
+} // namespace detail
+
+// clang-format off
+template<typename First, typename Second>
+concept ConvertibleRank = Rank<First>
+                          && Rank<Second>
+                          && detail::RechabilityHelper<Second, First>::value;
+// clang-format on
+
+} // namespace concepts
+
+template<concepts::Rank RankDefinition>
+class Location : public RankDefinition::Tuple {
+  friend RankDefinition;
+
+private:
+  using Tuple = typename RankDefinition::Tuple;
+  static_assert(std::tuple_size_v<Tuple> == RankDefinition::Depth);
+  static constexpr std::size_t Size = RankDefinition::Depth;
+
+private:
+  constexpr static std::string_view Separator = "/";
+
+public:
+  using Tuple::Tuple;
+  Location(const Location &) = default;
+  Location(Location &&) = default;
+  Location &operator=(const Location &) = default;
+  Location &operator=(Location &&) = default;
+
+  Tuple &tuple() { return *this; }
+  const Tuple &tuple() const { return *this; }
+
+  template<concepts::ConvertibleRank<RankDefinition> AnotherRank>
+  auto &get(const AnotherRank &Rank) {
+    return std::get<AnotherRank::Depth - 1>(tuple());
+  }
+  template<concepts::ConvertibleRank<RankDefinition> AnotherRank>
+  const auto &get(const AnotherRank &Rank) const {
+    return std::get<AnotherRank::Depth - 1>(tuple());
+  }
+
+  // clang-format off
+  template<typename AnotherRank>
+    requires (concepts::ConvertibleRank<RankDefinition, AnotherRank>
+              || concepts::ConvertibleRank<AnotherRank, RankDefinition>)
+  static constexpr Location<RankDefinition>
+  convert(Location<AnotherRank> &&Another) {
+    // clang-format on
+    Location<RankDefinition> Result;
+
+    constexpr auto Common = std::min(RankDefinition::Depth, AnotherRank::Depth);
+    constexprRepeat<Common>([&Result, &Another]<std::size_t I> {
+      std::get<I>(Result.tuple()) = std::move(std::get<I>(Another.tuple()));
+    });
+
+    return Result;
+  }
+
+  // clang-format off
+  template<typename AnotherRank>
+    requires (concepts::ConvertibleRank<RankDefinition, AnotherRank>
+              || concepts::ConvertibleRank<AnotherRank, RankDefinition>)
+  static constexpr Location<RankDefinition>
+  convert(const Location<AnotherRank> &Another) {
+    // clang-format on
+    Location<RankDefinition> Result;
+
+    constexpr auto Common = std::min(RankDefinition::Depth, AnotherRank::Depth);
+    constexprRepeat<Common>([&Result, &Another]<std::size_t I> {
+      std::get<I>(Result.tuple()) = std::get<I>(Another.tuple());
+    });
+
+    return Result;
+  }
+
+public:
+  std::string toString() const {
+    std::string Result;
+
+    (Result += Separator) += RankDefinition::RankName;
+    constexprRepeat<Size>([&Result, this]<auto Index> {
+      (Result += Separator) += serializeToString(std::get<Index>(tuple()));
```

Comment:
Split `Result += Separator` on their own line.

Reply:
Somehow I knew you're gonna say that 😅 Will do.

Reply:
Done: https://github.com/revng/revng/pull/197/commits/c7773a697b4c9eb46e98dbb9b28f1d3f7afaf40d